### PR TITLE
Refine docstring for 'compactness' kwarg in SLIC

### PR DESCRIPTION
**Pull request by jni**
Addresses #1745

### CODE REVIEW

File: skimage/segmentation/slic_superpixels.py [hunk 1]
```diff
@@ -25,10 +25,13 @@ def slic(image, n_segments=100, compactness=10., max_iter=10, sigma=0,
...
     compactness : float, optional
-        Balances color-space proximity and image-space proximity. Higher
-        values give more weight to image-space. As `compactness` tends to
-        infinity, superpixel shapes become square/cubic. In SLICO mode, this
-        is the initial compactness.
+        Balances color proximity and space proximity. Higher values give
+        more weight to space proximity, making superpixel shapes more
+        square/cubic. In SLICO mode, this is the initial compactness.
+        This parameter depends strongly on image contrast and on the
+        shapes of objects in the image; we recommend exploring possible
```
Comment (JDWarner) [inline skimage/segmentation/slic_superpixels.py]: I'd consider breaking this at the semicolon. Also, "We recommend initial experimentation with ``compactness`` on a log scale, e.g., 0.01, 0.1, 1, 10, 100" might be more concise.

**Comment by JDWarner** [comment]
Now all the 3.x builds are unhappy. I've raised a separate issue about these unrelated failures.

File: skimage/segmentation/slic_superpixels.py (hunk 1 above)
Comment (jni) [inline skimage/segmentation/slic_superpixels.py]: what do you mean by "breaking this at the semicolon"? Full stop or paragraph break? I wasn't sure about paragraph breaks in parameter descriptions. Advice?

Comment (JDWarner) [inline skimage/segmentation/slic_superpixels.py]: Break the sentence w/full stop, apologies for lack of clarity.

**Pull request merged**